### ISSUE TITLE
Texture UV mapping flips upside down.

### ISSUE THREAD

**Issue by dykarohora** [petrbroz/forge-convert-utils#18]
Hello,

When converting an FBX file exported from 3ds MAX to glTF, UV mapping was not processed correctly and it turned upside down. I found that the origin of the UV coordinate system in 3dsMAX is bottom-left, but the origin of the UV coordinate in glTF is in the top-left, so I guess this is the cause.

After modifying the code as follows, the UV mapping was processed correctly.
```
    // Parse zero or more UV maps
    const uvmaps: IUVMap[] = [];
    for (let i = 0; i < uvcount; i++) {
        name = pfr.getString(4);
        console.assert(name === 'TEXC');
        const uvmap: IUVMap = {
            name: '',
            file: '',
            uvs: new Float32Array()
        };
        uvmap.name = pfr.getString(pfr.getInt32());
        uvmap.file = pfr.getString(pfr.getInt32());
        uvmap.uvs = new Float32Array(vcount * 2);
        for (let j = 0; j < vcount * 2; j++) {
            // changed
            if (j % 2 === 0) {
                uvmap.uvs[j] = pfr.getFloat32();
            } else {
                uvmap.uvs[j] = 1.0 - pfr.getFloat32();
            }
        }
        uvmaps.push(uvmap);
```

However, the UV coordinate system may be different depending on the modeling software, so I think my modified code is not an appropriate modification. (Sadly I am not very familiar with modeling software...)

Do you have any idea to use the UV coordinate system of the original file properly?

**Comment by petrbroz** [comment]
Thanks for bringing this up @dykarohora.

Yes, in glTF 2.0 the UV origin (0,0) corresponds to the *upper-left corner* of the texture ([spec](https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/README.md#images)), and any rendering system that uses a different mapping is expected to post-process the UVs accordingly. For example, in case of OpenGL or WebGL where the UV origin is considered to be in the *bottom-left corner* of the texture, one would have to use the `gl.pixelStorei(gl.UNPACK_FLIP_Y_WEBGL, 1);` configuration.

With that being said, I have to admit that the support for UVs in this library is very basic, and it's possible that I'm missing some UV transformation there. Btw. if you view your FBX model directly in Forge Viewer, do the textures turn out ok? Or, if you wouldn't mind sharing one of your FBX files with me, I'd be happy to test/debug it myself.

**Comment by dykarohora** [comment]
[StandingTable.fbx.zip](https://github.com/petrbroz/forge-convert-utils/files/4062273/StandingTable.fbx.zip)

**Comment by dykarohora** [comment]
Sorry for replying late.
Share the fbx file that I use for testing.

The left one is converted with the version 3.0.0 code, and the right one is converted with the modified code.
<img width="1680" alt="cap" src="https://user-images.githubusercontent.com/6583336/72392926-29ce8600-36e6-11ea-92ac-a93146e3425f.png">

**Comment by petrbroz** [comment]
Thank you @dykarohora! Let me check the FBX model in Forge Viewer first, to see if the FBX-to-SVF pipeline converts everything properly.

**Comment by petrbroz** [comment]
So the Forge pipeline seems to convert the UVs properly as well:

![image](https://user-images.githubusercontent.com/1470668/72416168-5f22b480-3776-11ea-807b-329d2f078e06.png)

Good. Let me update the UV parsing logic based on your code then.

**Comment by petrbroz** [comment]
The fix is in. Will be available in version [3.1.0](https://github.com/petrbroz/forge-convert-utils/blob/develop/CHANGELOG.md#310---2020-01-15). Thanks again for the help!
